applets/clock/clock.c

```diff
@@ -901,8 +906,74 @@ position_calendar_popup (ClockData *cd)
         int             i, n;
         gboolean        found_monitor = FALSE;
 
+
         if (!GDK_IS_X11_DISPLAY (gdk_display_get_default ()))
-                return;
+        {
```

Comment:
This should be behind `GDK_IS_WAYLAND_DISPLAY` rather.  And not inside a `HAVE_X11` section, should it?

Reply:
It's actually a NOT X11 selector., but I can try switching that over, would be an needed if ever build without x11 at all

Reply:
GDK_IS_WAYLAND_DISPLAY worked fine, will push

Reply:
We now have both x11 and wayland ifdefs as well as the runtime selectors, as advised